src/dev/run_quick_checks.ts

```diff
@@ -0,0 +1,204 @@
+}) {
+  const { targetFile, targetDir, checks } = inputOptions;
+  if ([targetFile, targetDir, checks].filter(Boolean).length > 1) {
+    throw new Error('Only one of --file, --dir, or --checks can be used at a time.');
+  }
+
+  if (targetDir) {
+    const targetDirAbsolute = isAbsolute(targetDir) ? targetDir : join(REPO_ROOT, targetDir);
+    return readdirSync(targetDirAbsolute).map((file) => join(targetDir, file));
+  } else if (checks) {
+    return checks
+      .trim()
+      .split(/[,\n]/)
+      .map((script) => script.trim());
+  } else {
+    const targetFileWithDefault = targetFile || quickChecksList;
+    const targetFileAbsolute = isAbsolute(targetFileWithDefault)
+      ? targetFileWithDefault
+      : join(REPO_ROOT, targetFileWithDefault);
+
+    return readFileSync(targetFileAbsolute, 'utf-8')
+      .trim()
+      .split('\n')
+      .map((line) => line.trim());
+  }
+}
+
+async function runAllChecks(scriptsToRun: string[]) {
+  const checksRunning: Array<Promise<any>> = [];
+  const checksFinished: CheckResult[] = [];
+
+  while (scriptsToRun.length > 0 || checksRunning.length > 0) {
+    while (scriptsToRun.length > 0 && checksRunning.length < MAX_PARALLELISM) {
+      const script = scriptsToRun.shift();
+      if (!script) {
+        continue;
+      }
+
+      const check = runCheckAsync(script);
+      checksRunning.push(check);
+      check.then((result) => {
+        checksRunning.splice(checksRunning.indexOf(check), 1);
+        checksFinished.push(result);
+      });
+    }
+
+    await sleep(1000);
+  }
+
+  return checksFinished;
+}
+
+async function runCheckAsync(script: string): Promise<CheckResult> {
+  logger.info(`Starting check: ${script}`);
+  const startTime = Date.now();
+
+  return new Promise((resolve) => {
+    const scriptProcess = exec(script);
+    let output = '';
+    const appendToOutput = (data: string | Buffer) => (output += data);
+
+    scriptProcess.stdout?.on('data', appendToOutput);
+    scriptProcess.stderr?.on('data', appendToOutput);
+
+    scriptProcess.on('exit', (code) => {
+      const result: CheckResult = {
+        success: code === 0,
+        script,
+        output,
+        durationMs: Date.now() - startTime,
+      };
+      if (code === 0) {
+        logger.info(`Passed check: ${script} in ${humanizeTime(result.durationMs)}`);
+      } else {
+        logger.warning(`Failed check: ${script} in ${humanizeTime(result.durationMs)}`);
+      }
+
+      resolve(result);
+    });
+  });
+}
+
+function printResults(startTimestamp: number, results: CheckResult[]) {
+  const totalDuration = results.reduce((acc, result) => acc + result.durationMs, 0);
+  const total = humanizeTime(totalDuration);
+  const effective = humanizeTime(Date.now() - startTimestamp);
+  logger.info(`- Total time: ${total}, effective: ${effective}`);
+
+  results.forEach((result) => {
+    logger.write(
+      `--- ${result.success ? '✅' : '❌'} ${result.script}: ${humanizeTime(result.durationMs)}`
```

Comment:
Should we add some padding here (like an extra `-`) to make the logs more readable? (the idea is to simplify the hierarchy reading of the longs when failures happens)

Reply:
Does that make the logs on buildkite collapsible to multiple levels?

Reply:
I'll revert back to not adding an extra -, with `----` these sections are no longer collapsible.

If we stick with the `--- ✅` | `--- ❌` we'll have a high level print of the checks (successful or unsuccessful) at the end, and any issues can be opened/collapsed individually on buildkite, which I guess makes for a better reading experience. I'll attach images in a minute to show the alternatives.

Reply:
With `---`
<img width="761" alt="Screenshot 2024-08-21 at 10 12 04" src="https://github.com/user-attachments/assets/28e3c44b-eb77-4bfe-a3fb-7d2a282c88ca">
With `----`
<img width="768" alt="Screenshot 2024-08-21 at 09 54 12" src="https://github.com/user-attachments/assets/21ca9595-40f3-4862-8cbe-8b7a5087fe55">

Reply:
In that case lets proceed as it is

Reply:
this is quite an old error case, a lot of formatting changed since then. I'll try to produce an error again to see how it looks now

Reply:
@delanni the last iteration seems okay to me 👍